spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/SparkCatalogType.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark;
+
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+
+public enum SparkCatalogType {
+  TEST_HIVE("testhive", SparkCatalog.class.getName(), ImmutableMap.of(
+      "type", "hive",
+      "default-namespace", "default"
+  )),
+  TEST_HADOOP("testhadoop", SparkCatalog.class.getName(), ImmutableMap.of(
+      "type", "hadoop"
+  )),
+  SPARK_CATALOG("spark_catalog", SparkSessionCatalog.class.getName(), ImmutableMap.of(
+      "type", "hive",
+      "default-namespace", "default",
+      "parquet-enabled", "true",
+      "cache-enabled", "false" // Spark will delete tables using v1, leaving the cache out of sync
+  ));
```

Comment:
If we had constructors that provided a copy or something, so we had `SparkCatalogType.testHive(ImmutableMap.of("additionalConfig1", "additionalValue1")`, this would have much more benefit for the tests that need to override the `Parameters` to specify different catalog configuration.

Though that could be a v2 task.

Reply:
Thanks for your reply. 
Do you mean that the developers to specify only the additional config, without having to redeclare the entire catalog repeatedly?

SparkSpecifyCatalogTestBase provides the constructor as shown below, it can be used to add additional configuration items.

https://github.com/apache/iceberg/blob/9d431e87309275ecaf70c4a4b77f76a3dd921b90/spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/SparkSpecifyCatalogTestBase.java#L72-L79

Reply:
`super(SparkCatalogType.SPARK_CATALOG, ImmutableMap.of("additionalConfig1", "additionalValue1");`
Is it possible to use this method to achieve the desired results?